File: 1333-filter-restaurants-by-vegan-friendly-price-and-distance/1333-filter-restaurants-by-vegan-friendly-price-and-distance.java
class Solution {
    public List<Integer> filterRestaurants(int[][] rest, int v, int maxP, int maxD) {
        List<Integer>res=new ArrayList<>();
        Arrays.sort(rest, (a, b) -> {
            if (b[1]==a[1]){
                return Integer.compare(b[0], a[0]);
            }
            return Integer.compare(b[1], a[1]);
        });
        for (int[] x : rest){
            if (v==1 && x[2]==0){
                continue;
            } else if(x[3]>maxP || x[4]>maxD){
                continue;
            }
            res.add(x[0]);
        }
        
        return res;
    }
}

File: 1333-filter-restaurants-by-vegan-friendly-price-and-distance/1333-filter-restaurants-by-vegan-friendly-price-and-distance.cpp
class Solution {
public:
    vector<int> filterRestaurants(vector<vector<int>>& rest, int v, int maxP, int maxD) {
        sort(rest.begin(),rest.end(), [](auto& x, auto& y){
            if (x[1]==y[1]){
                return x[0]>y[0];
            }
            return x[1]>y[1];
        });
        vector<int>res;
        for (auto x : rest){
            if (v==1 && x[2]==0){
                continue;
            } else if(x[3]>maxP || x[4]>maxD){
                continue;
            }
            res.push_back(x[0]);
        }
        return res;
       
    }
};